src/main/kotlin/net/lumalyte/lg/infrastructure/services/ProgressionServiceBukkit.kt

```
@@ -111,6 +111,84 @@ class ProgressionServiceBukkit(
+            progressionRepository.recordExperienceTransaction(
+                ExperienceTransaction(
+                    guildId = guildId,
+                    source = source,
+                    amount = -amount,
+                    description = "Strike penalty: removed $amount XP"
+                )
+            )
+            syncGuildLevelField(guildId, newLevel)
+            logger.info("Guild $guildId XP reduced by $amount -> level $newLevel (strike penalty)")
+            return newLevel
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)